avogadro/rendering/geometryvisitor.h

```diff
@@ -61,7 +71,34 @@ class GeometryVisitor : public Visitor
    * Get the radius of the scene.
    */
   float radius();
+#ifdef _3DCONNEXION
+  /**
+   * <<API Extension for TDX>>
+   * Calculates the bounding box of the molecule.
+   * @param minX [out] minimum X coordinate of the box diagonal
+   * @param minY [out] minimum Y coordinate of the box diagonal
+   * @param minZ [out] minimum Z coordinate of the box diagonal
+   * @param maxX [out] maximum X coordinate of the box diagonal
+   * @param maxY [out] maximum Y coordinate of the box diagonal
+   * @param maxZ [out] maximum Z coordinate of the box diagonal
+   * @param flags [in] flags informing which atoms will be included
+   * in the bounding box.
+   */
+  void boundingBox(double& minX, double& minY, double& minZ, double& maxX,
+                   double& maxY, double& maxZ,
+                   const std::vector<bool>& flags) const;
 
+  /**
```

Comment:
Again. This seems like a reasonable API addition. Add it with no `#ifdef` and then the appropriate implementation.